api/naming_service.go

```diff
@@ -0,0 +1,110 @@
+package api
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"encoding/hex"
+
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/viper"
+	CoinType "github.com/trustwallet/blockatlas/coin"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	ens "github.com/wealdtech/go-ens/v3"
+)
+
+type ZNSResponse struct {
+	Addresses map[string]string
+	Meta      struct {
+		Owner string `json:"owner"`
+		Type  string `json:"type"`
+	} `json:"meta"`
+}
+
+// @Summary Lookup .eth / .zil addresses
+// @ID lookup
+// @Description Lookup ENS/ZNS to find registered addresses
+// @Produce json
+// @Tags ns
+// @Param name query string empty "string name"
+// @Param coin query string 60 "string coin"
+// @Success 200 {object} blockatlas.Resolved
+// @Failure 500 {object} api.ApiError
+// @Router /ns/lookup [get]
+func MakeLookupRoute(router gin.IRouter) {
+	ns := router.Group("/ns")
+	ns.GET("/lookup", handleLookup)
+}
+
+func handleLookup(c *gin.Context) {
+	name := c.Query("name")
+	coinQuery := c.DefaultQuery("coin", strconv.Itoa(CoinType.ETH))
+
+	if name == "" {
+		RenderError(c, http.StatusBadRequest, "name query is missing")
+		return
+	}
+	coin, err := strconv.ParseUint(coinQuery, 10, 64)
+	if err != nil {
+		RenderError(c, http.StatusBadRequest, "coin query is invalid")
+		return
+	}
+	if strings.Contains(name, ".eth") {
```

Comment:
should we check for suffix? I think if you type test.eth.zil it will trigger as true.